File: AddressBook/AddressBook/Dtos/PersonChat.cs
using AddressBook.Entities;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace AddressBook.Dtos
{
    public class PersonChatRes
    {
        public int PersonChatId { get; set; }
        public string? Chat { get; set; }

        public int? PersonId { get; set; }
        public PersonRes? Person { get; set; }

        public int? ChatLabelId { get; set; }
        public ChatLabelRes? ChatLabel { get; set; }
    }

    public class PersonChatReqEdit
    {
        [Required]
        public string? Chat { get; set; }
        public int? PersonId { get; set; }
        public int? ChatLabelId { get; set; }
    }

    public class PersonChatReqSearch
    {
        public int PersonId { get; set; }
    }
}

File: AddressBook/react-client/src/dtos/PersonChat.ts
import Common from "../utility/Common";
import { ChatLabelRes } from "./ChatLabel";
import { PagedReq } from "./PagedReq";
import { PersonRes } from "./Person";

export interface PersonChatRes {
  personChatId?: number;
  chat?: string;
  personId?: number;
  person?: PersonRes;
  chatLabelId?: number;
  chatLabel?: ChatLabelRes;
}

export class PersonChatReqEdit {
  chat?: string = "";
  personId?: number = 0;
  chatLabelId?: number = 0;
}

export class PersonChatReqSearch extends PagedReq {
  personId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {personId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.personId = personId;
  }
}